src/foundry/foundry.js/clientDocuments/chatMessage.d.ts

```diff
@@ -1,16 +1,221 @@
+     * Alternatively it could be a User name in the case of OOC chat or whispers.
+     */
+    get alias(): string;
+
+    /**
+     * Is the current User the author of this message?
+     */
+    get isAuthor(): boolean;
+
+    /**
+     * Return whether the content of the message is visible to the current user.
+     * For certain dice rolls, for example, the message itself may be visible while the content of that message is not.
+     */
+    get isContentVisible(): boolean;
+
+    /**
+     * Test whether the chat message contains a dice roll
+     */
+    get isRoll(): boolean;
+
+    /**
+     * Return the Roll instance contained in this chat message, if one is present
+     */
+    get roll(): Roll | null;
+
+    /**
+     * Return whether the ChatMessage is visible to the current User.
+     * Messages may not be visible if they are private whispers.
+     */
+    get visible(): boolean;
+
+    /**
+     * The User who created the chat message.
+     */
+    get user(): User | undefined;
+
+    /** @override */
+    prepareData(): void;
+
+    /**
+     * Transform a provided object of ChatMessage data by applying a certain rollMode to the data object.
+     * @param chatData - The object of ChatMessage data prior to applying a rollMode preference
+     * @param rollMode - The rollMode preference to apply to this message data
+     * @returns The modified ChatMessage data with rollMode preferences applied
+     */
+    static applyRollMode(
+      chatData: foundry.data.ChatMessageData,
+      rollMode: foundry.CONST.DiceRollMode
+    ): foundry.data.ChatMessageData;
+
+    /**
+     * Update the data of a ChatMessage instance to apply a requested rollMode
+     * @param rollMode - The rollMode preference to apply to this message data
+     */
+    applyRollMode(rollMode: foundry.CONST.DiceRollMode): void;
+
+    /**
+     * Attempt to determine who is the speaking character (and token) for a certain Chat Message
+     * First assume that the currently controlled Token is the speaker
+     *
+     * @param scene - The Scene in which the speaker resides
+     *                (defaultValue: `undefined`)
+     * @param actor - The Actor whom is speaking
+     *                (defaultValue: `undefined`)
+     * @param token - The Token whom is speaking
+     *                (defaultValue: `undefined`)
+     * @param alias - The name of the speaker to display
+     *                (defaultValue: `undefined`)
+     *
+     * @returns The identified speaker data
+     */
+    static getSpeaker({
+      scene,
+      actor,
+      token,
+      alias
+    }?: {
+      scene?: InstanceType<ConfiguredDocumentClass<typeof Scene>>;
+      actor?: InstanceType<ConfiguredDocumentClass<typeof Actor>>;
+      token?:
+        | InstanceType<ConfiguredDocumentClass<typeof TokenDocument>>
+        | InstanceType<CONFIG['Token']['objectClass']>;
+      alias?: string;
+    }): ChatSpeakerData;
```

Comment:
The return type is incorrect, this doesn't actually return an instance of that class. Instead, I think what is returned is probably the corresponding _source type.